Infrastructure: Size report action failing

**Expected behavior**

The size report action completes successfully (even if the validation fails)

**Actual behavior**

The size report errors out

**Steps to reproduce the issue**

1. Open a pull request
2. Observe the size action error

**Screenshots or code**

[Failing build example](https://github.com/ithaka/pharos/runs/6277629095?check_suite_focus=true)

<details>
<summary>Error output</summary>

```
Run andresz1/size-limit-action@v1
/opt/hostedtoolcache/node/16.14.2/x64/bin/npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: pharos@0.0.1
npm ERR! Found: @storybook/addons@6.5.0-alpha.5
npm ERR! node_modules/@storybook/addons
npm ERR!   dev @storybook/addons@"^6.5.0-alpha.5" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @storybook/addons@"^6.1.14" from @storybook/addon-google-analytics@6.3.0-next.1
npm ERR! node_modules/@storybook/addon-google-analytics
npm ERR!   dev @storybook/addon-google-analytics@"^6.3.0-alpha.22" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/runner/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2022-05-03T17_28_55_984Z-debug-0.log
Error: The process '/opt/hostedtoolcache/node/16.14.2/x64/bin/npm' failed with exit code 1
```

</details>

--- a/.changeset/new-cameras-prove.md
+++ b/.changeset/new-cameras-prove.md
@@ -1,0 +1,5 @@
+---
+'@ithaka/pharos': patch
+---
+
+Move size-limit checks up to the project root

--- a/.github/workflows/size-limit.yml
+++ b/.github/workflows/size-limit.yml
@@ -18,9 +18,6 @@ jobs:
         with:
           node-version: 16.x
 
-      - name: Install Yarn
-        run: npm install --global yarn
-
       - name: Get yarn cache directory path
         id: yarn-cache-dir-path
         run: echo "::set-output name=dir::$(yarn cache dir)"
@@ -38,4 +35,4 @@ jobs:
       - uses: andresz1/size-limit-action@v1
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
-          directory: packages/pharos/
+          build_script: 'build:core'

--- a/.github/workflows/size-limit.yml
+++ b/.github/workflows/size-limit.yml
@@ -18,6 +18,9 @@ jobs:
         with:
           node-version: 16.x
 
+      - name: Install Yarn
+        run: npm install --global yarn
+
       - name: Get yarn cache directory path
         id: yarn-cache-dir-path
         run: echo "::set-output name=dir::$(yarn cache dir)"

--- a/.github/workflows/size-limit.yml
+++ b/.github/workflows/size-limit.yml
@@ -18,20 +18,6 @@ jobs:
         with:
           node-version: 16.x
 
-      - name: Get yarn cache directory path
-        id: yarn-cache-dir-path
-        run: echo "::set-output name=dir::$(yarn cache dir)"
-
-      - uses: actions/cache@v2
-        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
-        with:
-          path: |
-            ${{ steps.yarn-cache-dir-path.outputs.dir }}
-            **/node_modules
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
-
       - uses: andresz1/size-limit-action@v1
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}

--- a/.github/workflows/size-limit.yml
+++ b/.github/workflows/size-limit.yml
@@ -18,6 +18,20 @@ jobs:
         with:
           node-version: 16.x
 
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - uses: actions/cache@v2
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: |
+            ${{ steps.yarn-cache-dir-path.outputs.dir }}
+            **/node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
       - uses: andresz1/size-limit-action@v1
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}

--- a/package.json
+++ b/package.json
@@ -64,6 +64,7 @@
     "@changesets/cli": "^2.18.0",
     "@commitlint/cli": "^15.0.0",
     "@commitlint/config-conventional": "^15.0.0",
+    "@size-limit/preset-small-lib": "^7.0.4",
     "@storybook/addon-a11y": "^6.5.0-alpha.5",
     "@storybook/addon-actions": "^6.5.0-alpha.5",
     "@storybook/addon-backgrounds": "^6.5.0-alpha.5",
@@ -111,6 +112,7 @@
     "react-dom": "^17.0.1",
     "sass": "^1.42.1",
     "sass-loader": "^12.3.0",
+    "size-limit": "^7.0.4",
     "style-loader": "^3.3.1",
     "stylelint": "^14.0.0",
     "stylelint-config-prettier": "^9.0.3",
@@ -141,5 +143,10 @@
     "trim": "^0.0.3",
     "marked": "^0.7.0",
     "graphql-config": "^4.1.0"
-  }
+  },
+  "size-limit": [
+    {
+      "path": "packages/pharos/lib/index.js"
+    }
+  ]
 }

--- a/packages/pharos/package.json
+++ b/packages/pharos/package.json
@@ -63,7 +63,6 @@
   "devDependencies": {
     "@custom-elements-manifest/analyzer": "^0.5.6",
     "@open-wc/testing": "^3.0.1",
-    "@size-limit/preset-small-lib": "^7.0.4",
     "@types/react-dom": "^17.0.0",
     "@types/react": "^17.0.1",
     "@web/test-runner": "^0.13.16",
@@ -82,7 +81,6 @@
     "sass": "^1.42.1",
     "sassdoc": "^2.7.3",
     "sinon": "^12.0.1",
-    "size-limit": "^7.0.4",
     "style-dictionary": "^3.0.1",
     "ts-lit-plugin": "^1.2.1",
     "typescript": "^4.5.2"

--- a/packages/pharos/package.json
+++ b/packages/pharos/package.json
@@ -85,10 +85,5 @@
     "ts-lit-plugin": "^1.2.1",
     "typescript": "^4.5.2"
   },
-  "customElements": "custom-elements.json",
-  "size-limit": [
-    {
-      "path": "lib/index.js"
-    }
-  ]
+  "customElements": "custom-elements.json"
 }

